barcodes_search/wizards/barcode_search.py

```diff
@@ -0,0 +1,121 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2017-Today GRAP (http://www.grap.coop)
+# @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class BarcodeSearch(models.TransientModel):
+    _name = 'barcode.search'
+
+    barcode = fields.Char(
+        string='Barcode', required=True, help='Barcode to search for.')
+
+    state = fields.Selection(
+        selection=[('new', 'New'), ('done', 'Done')], default='new')
+
+    line_ids = fields.One2many(
+        comodel_name='barcode.search.line', inverse_name='barcode_search_id',
+        readonly=True)
+
+    @api.multi
+    def button_search(self):
+        self.ensure_one()
+        res = self.search_by_barcode(self.barcode)
+        if len(res) == 1 and not res[0][2]:
+            # Display the result in the according form view
+            record = res[0][1]
+            return record.get_formview_action()
+        else:
+            # Update Wizard Values
+            line_vals = []
+            self.line_ids.unlink()
```

Comment:
IMO it would be better to put a `(5, 0)` in `line_vals` declaration and drop this line to save a one-off db call.

From what I understand, the records on O2M are cleaned when the foreign key is a required field. In this instance regardless, the vacuum will hit them.